vnavmesh/Navmesh.cs

```diff
@@ -24,9 +24,17 @@ public static Navmesh Deserialize(BinaryReader reader, int expectedCustomization
         if (customizationVersion != expectedCustomizationVersion)
             throw new Exception("Outdated customization version");
 
-        using var compressedReader = new BinaryReader(new BrotliStream(reader.BaseStream, CompressionMode.Decompress, true));
-        var mesh = DeserializeMesh(reader);
-        var volume = DeserializeVolume(reader);
+        var compressedDataLength = reader.ReadInt32();
+        //Service.Log.Debug($"[Deserialize] Header, Magic {magic:X} , Version {version}, cv {customizationVersion}, length {compressedDataLength}");
+        byte[] compressedData = reader.ReadBytes(compressedDataLength);
+
+        using var compressedStream = new MemoryStream(compressedData);
+        using var decompressedStream = new BrotliStream(compressedStream, CompressionMode.Decompress, true);
+        using var decompressedReader = new BinaryReader(decompressedStream);
+
+        var mesh = DeserializeMesh(decompressedReader);
+        var volume = DeserializeVolume(decompressedReader);
```

Comment:
Oh wow, that's a good catch - so (de)compress streams were created, but not used, lol.

Is there a reason why you did it this way (with extra memory buffer) rather than just passing compressedReader / compressedWriter to the (de)serialize functions?

Reply:
Just tried it locally and it seems to work as well btw...

Reply:
ah, good catch, i believe i had that before files were truncated to prevent errors